wecs/core.py

```diff
@@ -409,6 +409,54 @@ def __call__(self, cls):
         return cls
 
 
+class Proxy:
+    """
+    When at coding time it is not yet known what component types and
+    field names will be used for an aspect of functionality that a
+    system will work on (e.g. where to find a `NodePath` to center a
+    camera on), a `Proxy` can be used instead to declare a name under
+    which the definition can actually be looked up. For example:
+
+        class ProxyingSystem(System):
+            entity_filters = {
+                'test': Proxy('proxy'),
+            }
+    
+            def update(self, entity_by_filters):
+                for entity in entity_by_filters['test']:
+                    proxy = self.proxies['proxy']
+                    my_component = entity[proxy.component_type]
+    
+                    field_value = proxy.field(my_component)
+    
+
+        @Component()
+        class MyComponent:
+            foo: str = '123'
+
+
+        class ActualSystem(ProxyingSystem):
+            proxies = {
+                'proxy': ProxyType(MyComponent, 'foo'),
+            }
+
+    `ActualSystem` thus is an implementation of `ProxyingSystem` where
+    `my_component = entity[MyComponent]` and 
+    `field_value = my_component.foo`.
+    """
+    def __init__(self, name):
+        self.name = name
+
+
+class ProxyType:
+    def __init__(self, component_type, field_name=None):
+        self.component_type = component_type
+        self.field_name = field_name
+
+    def field(self, entity):
+        component = entity[self.component_type]
+        return getattr(component, self.field_name)
+
```

Comment:
didn't really debug or tested, but I have a feeling both of these can be refactored to a separate file.

Reply:
I'm afraid I don't quite understand? What "both"? A separate file for what?